File: backend-spring/src/main/java/com/sourcelens/module/review/entity/PrReview.java
package com.sourcelens.module.review.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("pr_reviews")
public class PrReview {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long projectId;

    private Long scanTaskId;

    private Long repositoryId;

    private Integer prNumber;

    private String prTitle;

    private String prDescription;

    private String branch;

    private String baseBranch;

    private String commitSha;

    private String author;

    private String changedFiles;

    private String diffSummary;

    private String ciStatus;

    /** PENDING / ANALYZING / COMPLETED / FAILED */
    private String status;

    /** LOW / MEDIUM / HIGH / CRITICAL */
    private String riskLevel;

    private String changeSummary;

    private String impactScope;

    private String risks;

    private String testSuggestions;

    /** MERGE / CHANGES_REQUESTED / BLOCKED */
    private String mergeRecommendation;

    private String reviewJson;

    private String errorMessage;

    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Boolean deleted;
}

File: web-console/src/api/prReview.ts
import client from './client'
import type { Result } from './client'

export interface PrReview {
  id: number
  projectId: number
  scanTaskId: number | null
  repositoryId: number | null
  prNumber: number | null
  prTitle: string | null
  prDescription: string | null
  branch: string | null
  baseBranch: string | null
  commitSha: string | null
  author: string | null
  changedFiles: string | null
  diffSummary: string | null
  ciStatus: string | null
  status: string
  riskLevel: string | null
  changeSummary: string | null
  impactScope: string | null
  risks: string | null
  testSuggestions: string | null
  mergeRecommendation: string | null
  reviewJson: string | null
  errorMessage: string | null
  createdBy: number
  createdAt: string
  updatedAt: string
}

export interface PrReviewComment {
  id: number
  reviewId: number
  filePath: string
  lineNumber: number | null
  severity: string
  category: string
  message: string
  suggestion: string | null
  createdAt: string
}

export interface PageResult<T> {
  items: T[]
  page: number
  pageSize: number
  total: number
}

export const prReviewApi = {
  create: (data: {
    projectId: number
    scanTaskId?: number
    repositoryId?: number
    prNumber?: number
    prTitle?: string
    prDescription?: string
    branch?: string
    baseBranch?: string
    commitSha?: string
    author?: string
    changedFiles?: string
    diffSummary?: string
    ciStatus?: string
  }) => client.post<Result<PrReview>>('/pr-reviews', data),

  listByProject: (projectId: number, page = 1, pageSize = 20, status?: string) => {
    const params = new URLSearchParams({ page: String(page), pageSize: String(pageSize) })
    if (status) params.set('status', status)
    return client.get<Result<PageResult<PrReview>>>(`/projects/${projectId}/pr-reviews?${params}`)
  },

  detail: (id: number) =>
    client.get<Result<PrReview>>(`/pr-reviews/${id}`),

  listComments: (id: number) =>
    client.get<Result<PrReviewComment[]>>(`/pr-reviews/${id}/comments`),

  reanalyze: (id: number) =>
    client.post<Result<PrReview>>(`/pr-reviews/${id}/reanalyze`),
}